### PR TITLE
Fix fundies broken DEBUG_MESSAGE macro

### PR DESCRIPTION
**Pull request by time-killer-games**
Makes it only show in debug mode. Fixes https://github.com/enigma-dev/enigma-dev/issues/1773

Screenshot_20190704_204127
Screenshot_20190704_204144
Screenshot_20190704_204205

### CODE REVIEW

**Comment by time-killer-games** [comment]
@JoshDreamland Alrighty, It's ready for review when you are. Thanks!

**Comment by time-killer-games** [comment]
@fundies I needed message_get/set_caption() for setting the appropriate caption so that it identifies the severity string. I use message_get_caption() to set it back to the custom caption set by the game developer for when they want to use their own dialogs and captions.

"wrong fix" doesn't tell me anything if you don't say what's wrong with it.

**Comment by fundies** [comment]
read above. ive edited it. The code is not broken and is working as intended. you need to make changes to your implementation of show_debug_message to print what you want and to mask any useless info / errors we currently print in the engine

**Comment by time-killer-games** [comment]
@fundies alright dude taking care of it as we speak :)

File: ENIGMAsystem/SHELL/Widget_Systems/KDialog/dialogs.cpp [hunk 1]
```diff
@@ -196,6 +192,20 @@ void show_debug_message(string errortext, MESSAGE_TYPE type) {
   if (strtod(str_result.c_str(), NULL) == 1) exit(0);
 }
 
+namespace enigma_user {
+
+void show_debug_message(string errortext, MESSAGE_TYPE type) {
+  if (type != M_INFO && type != M_WARNING) {
+    #ifndef DEBUG_MODE
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Widget_Systems/KDialog/dialogs.cpp]: Let the caller decide if they want special handling for debug mode, Sam. People will want to know that the graphics system failed to initialize, even outside of debug mode.

**Comment by RobertBColton** [comment]
@time-killer-games This was a good fix, but you copied the exact same code to 5~ places. Code duplication is generally avoided because it increases the chances of bugs.

**Pull request merged**